docs/src/pages/components/tooltips/AnchorElTooltips.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import Box from '@material-ui/core/Box';
+import Tooltip from '@material-ui/core/Tooltip';
+import PopperJs from 'popper.js';
+
+export default function AnchorElTooltips() {
+  const positionRef = React.useRef<{ x: number; y: number }>({
+    x: 0,
+    y: 0,
+  });
+  const popperRef = React.useRef<PopperJs>(null);
+  const areaRef = React.useRef<HTMLDivElement>(null);
+
+  const handleMouseMove = (event: React.MouseEvent) => {
+    positionRef.current = { x: event.clientX, y: event.clientY };
+
+    if (popperRef.current != null) {
+      popperRef.current.scheduleUpdate();
+    }
+  };
+
+  return (
+    <Tooltip
+      title="Add"
+      placement="top"
+      arrow
+      PopperProps={{
+        popperRef,
+        anchorEl: {
+          clientHeight: 0,
+          clientWidth: 0,
+          getBoundingClientRect: () => ({
+            top: areaRef.current?.getBoundingClientRect().top ?? 0,
+            left: positionRef.current.x,
+            right: positionRef.current.x,
+            bottom: areaRef.current?.getBoundingClientRect().bottom ?? 0,
+            width: 0,
+            height: 0,
+          }),
+        },
+      }}
+    >
+      <div ref={areaRef}>
+        <Box
```

Comment:
I couldn't use ref because of #17010

Reply:
If it's types only then prefer an `as any` cast.

Reply:
OK, I will replace it with an any.

Reply:
`@ts-expect-error` was the better choice here. See https://github.com/mui-org/material-ui/pull/22876#discussion_r500983732.